### PR TITLE
Remove kind tests running on Prow

### PR DESCRIPTION
**Pull request by JustinKuli**
These tests now run in GitHub Actions, and can be removed here.

Refs:
 - https://github.com/stolostron/backlog/issues/22914
 - https://github.com/stolostron/governance-policy-framework/pull/455

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

### CODE REVIEW

**Review by dhaiducek — CHANGES_REQUESTED** [review]
Would you please also update the branch protection rules here and double check the `only_if_changed` in the job so that the E2E doesn't run when the workflows are updated? (I'll hunt down the location of that latter one...)

File: ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-main.yaml [hunk 1]
```diff
@@ -30,45 +30,6 @@ tests:
...
-      - env: IMAGE_REF
-        name: grc-policy-framework-tests
-      from: src
-      resources:
-        requests:
-          cpu: 100m
-          memory: 200Mi
-    workflow: ocm-e2e-kind
 - as: test-e2e-grc-framework
   run_if_changed: ^test/integration/.*
```
Comment (JustinKuli) [inline ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-main.yaml:34]: @dhaiducek is this the bit you meant?

Comment (dhaiducek) [reply ci-operator/config/stolostron/governance-policy-framework/stolostron-governance-policy-framework-main.yaml:34]: Haha--yeah. It's definitely okay the way it is. 😄

**Review by dhaiducek — APPROVED** [review]
LGTM!

I suppose we might want to cherry-pick these to `2.6` and maybe `2.5` (and `2.4` if we're feeling super ambitious). Might as well save money everywhere...

**Pull request merged**